### PR TITLE
shaderObject: Fix null pointer dereference if privateData or dynamicRendering not supported

### PR DESCRIPTION
**Pull request by DHrpcs3**
See initialization flow: https://github.com/KhronosGroup/Vulkan-ExtensionLayer/blob/main/layers/shader_object/generated/shader_object_create_device_feature_structs.inl

### CODE REVIEW

**Review by ziga-lunarg — CHANGES_REQUESTED** [review]
This should be `!dynamic_rendering_ptr || dynamic_rendering_ptr->dynamicRendering == VK_FALSE`

Same below for private_data_ptr

File: layers/shader_object/shader_object.cpp [hunk 1]
```diff
@@ -2034,7 +2034,7 @@ static VKAPI_ATTR VkResult VKAPI_CALL CreateDevice(VkPhysicalDevice physicalDevi
 #include "generated/shader_object_create_device_feature_structs.inl"
         device_features.pNext = appended_features_chain;
         instance_vtable.GetPhysicalDeviceFeatures2(physicalDevice, &device_features);
-        if (dynamic_rendering_ptr->dynamicRendering == VK_FALSE) {
+        if (dynamic_rendering_ptr && dynamic_rendering_ptr->dynamicRendering == VK_FALSE) {
```
Comment (ziga-lunarg) [inline layers/shader_object/shader_object.cpp:2037]: Here

**Comment by DHrpcs3** [comment]
> Same below for private_data_ptr
Seems private_data_ptr is ok, because it is inverted check

**Review by ziga-lunarg — APPROVED** [review]

**Comment by ziga-lunarg** [comment]
I suggest you squash the commits together, the second commit message does not pass the format check.

**Comment by DHrpcs3** [comment]
Could you please press Squash and merge button and name resulting commit as you wish?
I have no local clone of this repository
Screenshot From 2024-09-16 17-56-05

**Comment by ziga-lunarg** [comment]
It's not enabled for this repository and we want all the checks to pass before we do that. I can however manually squash your commits.

**Pull request merged**